src/api/app/views/webui/project/_buildstatus.html.erb

```diff
@@ -21,13 +21,23 @@
                       </td>
                       <% index += 1 %>
                   <% end %>
-                  <td class="nowrap">
-                    <%= repo_status_icon(@repostatushash[repository][arch], @repostatusdetailshash[repository][arch])%> <%= arch %>
+                  <td class="arch" title="<%= "#{repository} summary" %>" >
+                    <div class="nowrap" style="margin: 0 0.5ex">
+                    <%= repo_status_icon(@repostatushash[repository][arch], @repostatusdetailshash[repository][arch])%> <%= link_to(arch, {
+                          :action => :monitor, valid_xml_id('repo_' + repository) => 1,
+                          'arch_' + arch => 1, :project => params[:project],
+                          :succeeded => 1, :failed => 1, :unresolvable => 1, :broken => 1,
```

Comment:
Yes, it looks pretty ugly.. The idea was to show same build states what are shown in the project monitor page by default (e.g. disabled, excluded and unknown states are not shown). I think it would show all states unless those specific states are set to 1.